### ISSUE TITLE
Reorder topology modules to satisfy module dependencies

### ISSUE THREAD

**Issue by ipspace** [ipspace/netsim-tools#86]
### Discussed in https://github.com/ipspace/netsim-tools/discussions/83

<div type='discussions-op-text'>

<sup>Originally posted by **jbemmel** December 21, 2021</sup>
It looks like modules are currently processed in the order declared. For example, from routing sr-mpls-bgp example:
```
nodes:
  e1:
    device: csr
    module: [ sr, isis, bgp ]
    bgp:
      as: 65000
```

'sr' depends on 'isis', but this order of deployment is currently not enforced(?)

Also, there can be a case where 'sr' modifies 'bgp' configuration - there would seem to be a preferred execution order:
1. isis (or ospf)
2. bgp (first ebgp, then ibgp)
3. sr

(in my case I tried to configure a BGP shortcut for SR, which fails because BGP AS is not configured)</div>

**Comment by ipspace** [comment]
Implemented in 5dd57b1

**Comment by ipspace** [comment]
Implemented in 5dd57b1
